pkg/lib/hooks.js

```diff
@@ -325,3 +325,29 @@ export function useEvent(obj, event, handler) {
 export function useInit(func, deps, comps) {
     return useObject(func, null, deps || [], comps);
 }
+
+/* XXX - document this
+ */
+
+export function useMockConfig() {
+    useEvent(window, "test_mock_changed");
+    return window.test_mock;
+}
+
+export function useMock(topic) {
+    const config = useMockConfig();
+
+    return (val, tag) => {
+        let mock = config;
+
+        if (!mock)
+            return val;
+
+        if (topic)
+            mock = mock[topic];
+        if (tag && mock)
+            mock = mock[tag];
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17654-20220819-083129-c0e512fa-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)